### ISSUE TITLE
[Android] PointerIcon leak in Cursor API

### ISSUE THREAD

**Issue by StephenCWills** [libsdl-org/SDL#5118]
The Cursor API on Android depends on a Java `Hashtable` to map NDK objects to JDK objects via an integer key marshalled through JNI calls.

https://github.com/libsdl-org/SDL/blob/398c0fdff0946ddb87853a474be286d3c01b89ca/android-project/app/src/main/java/org/libsdl/app/SDLActivity.java#L112

The only references made to this `Hashtable` in the Java codebase can be found in the following two methods that appear neatly map to `SDL_CreateCursor()` and `SDL_SetCursor()` calls.

https://github.com/libsdl-org/SDL/blob/398c0fdff0946ddb87853a474be286d3c01b89ca/android-project/app/src/main/java/org/libsdl/app/SDLActivity.java#L1463-L1497

In other words, the `Hashtable` only grows and never shrinks, effectively causing a memory leak. It seems to me there should be a JNI call for `SDL_FreeCursor()` as well so the `PointerIcon` objects can be removed from the `Hashtable` and claimed by the garbage collector.

**Comment by 1bsyl** [comment]
Thanks this is fixing it rigth ? once the object is remove from the hashtable it will be collected ?

**Comment by StephenCWills** [comment]
Yeah, the GC should take care of it at that point. Thanks for having a look.

**Comment by 1bsyl** [comment]
you're welcome thanks for reporting it !
